llvm/lib/CodeGen/MachineCopyPropagation.cpp

```diff
@@ -971,6 +971,18 @@ void MachineCopyPropagation::ForwardCopyPropagateBlock(MachineBasicBlock &MBB) {
 
     forwardUses(MI);
 
+    // It's possible that the previous transformation has resulted in a no-op
+    // register move (i.e. one where source and destination registers are the
+    // same). If so, delete it.
+    CopyOperands = isCopyInstr(MI, *TII, UseCopyInstr);
```

Comment:
Placement wise, I think this works better as an if before the existing !TRI->regsOverlap(RegDef, RegSrc) check specific for copies above.  Purely a stylistic comment.

Reply:
I'm a bit confused here - this logic needs to go after the forwardUses just above, so I don't think I have any flexibility to move it up to where you suggest?

Reply:
You're right, though in abstract we could organize the code to have forwardUses about the split point.  (Not asking you to do this.)